ls1intum/Artemis
# Sync sever and client time

### Checklist

### Motivation and Context


Students can manipulate their system time and thus gain an unfair advantage seeing the exam and the exercises before it starts.

### Description

With every request a header is sent with the current time/date of the server.
The ArtemisServerDateService stores this and provides the time 'in sync' with the server.
The exam mode uses this time instead of the system time to prevent cheating.

### Steps for Testing


1. Log in to Artemis
2. Setup a test exam
3. Check (especially) the following scenarios with and without manipulating your system time:
  -  wait until time starts
  - exam ends automatically
4. The remaining time should be displayed correctly

Make sure the exam mode is **immune** to system time changes
